### PR TITLE
refactor: using death state as a statebag

### PR DESCRIPTION
**Pull request by Manason**
- updating player's metadata for isDead and inLaststand when death state changes

### CODE REVIEW

File: server/main.lua [hunk 1]
```diff
@@ -16,12 +16,20 @@ local playerState
...
 	playerState = Player(source).state
+	playerState:set(DEATH_STATE_STATE_BAG, Config.DeathState.ALIVE, true)
 	playerState:set(BLEED_LEVEL_STATE_BAG, 0, true)
 	for bodyPartKey in pairs(Config.BodyParts) do
 		playerState:set(BODY_PART_STATE_BAG_PREFIX .. bodyPartKey, nil, true)
 	end
 end)
 
+AddStateBagChangeHandler(DEATH_STATE_STATE_BAG, nil, function(bagName, _, value)
+    local playerId = string.sub(bagName, 8)
```
Comment (FjamZoo) [inline server/main.lua:27]: You should use this native here: https://docs.fivem.net/natives/?_0xA56135E0

File: server/main.lua [hunk 2]
```diff
@@ -16,12 +16,20 @@ local playerState
 
 AddEventHandler('QBCore:Server:OnPlayerLoaded', function()
 	playerState = Player(source).state
+	playerState:set(DEATH_STATE_STATE_BAG, Config.DeathState.ALIVE, true)
```
Comment (FjamZoo) [inline server/main.lua:19]: Wouldn't we apply if the player was dead via the PlayerData from core here?

**Review by FjamZoo — CHANGES_REQUESTED** [review]
There's 1 change requested and 1 question :)

File: server/main.lua (hunk 1 above)
Comment (Manason) [reply server/main.lua]: done

File: server/main.lua (hunk 2 above)
Comment (Manason) [reply server/main.lua]: good point. Fixed to load from metadata.

**Review by FjamZoo — APPROVED** [review]

**Pull request merged**